[Jazer Ler] iP

jazerler opened the pull request.

Commit:
Add Gradle support

jazerler committed:
Add Greet, Echo, Exit

jazerler committed:
Add List and Add

jazerler committed:
Add mark task as done, change the reply format

jazerler committed:
Add Todos, Events, Deadlines

jazerler committed:
Add error handling

jazerler committed:
Add Delete functionality

jazerler committed:
Add automated text UI testing

jazerler committed:
Format parts of Duke.java

jazerler committed:
Add saving and reading to and from disk

Data is now saved under ./data.txt.

jazerler committed:
Add dates and times with LocalDate

jazerler committed:
Merge branch 'branch-Level-7' into master

jazerler committed:
Clear log before merge

jazerler committed:
Merge tag 'branch-Level-8' into master

jazerler committed:
Refactor code to extract more classes

jazerler committed:
Clear logs

jazerler committed:
Create duke package

jazerler committed:
Add packages

jazerler committed:
Add test directory

jazerler committed:
Add JUnit tests for Parser

jazerler committed:
Reformat code

jazerler committed:
Add JavaDoc comments

jazerler committed:
Rename variables to follow code style

jazerler committed:
Add FindCommand and ResultsList

jazerler committed:
Merge branch 'A-JavaDoc' into master

jazerler committed:
Merge branch 'A-CodingStandard' into master

jazerler committed:
Merge branch 'branch-Level-9' into master

jazerler committed:
Update coding style and Javadoc comments for FindCommand

jazerler committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2021S1/ip into add-gradle-support

jazerler committed:
Change folder structure and add Gradle

jazerler committed:
Add CheckStyle and reformat offending lines

jazerler committed:
Add sceneplayground package for JavaFX testing

Placeholder package for experimenting with the GUI.
Placeholder images also added.

jazerler committed:
Add GUI to Duke

Bugs:
Tick and Cross symbols don't display properly in GUI.
GUI does not format the replies properly as of yet.

jazerler committed:
Add colour to GUI

jazerler committed:
Merge branch 'branch-Level-10' into master

Nahoyhp commented on src/main/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Override
+    public boolean shouldExit() {
```
Any reason why you don't choose to omit the shouldExit() method? 
[Same for other command classes that return false as well]
From my understanding, if the sub-class behaves exactly as the parent class, we can choose to omit it and the system can compile.

jazerler replied:
Yes, you're right that it can be omitted. Will be removing this in the next commit, thanks.

Nahoyhp commented on src/main/duke/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    public String[] toArray() {
```
This seems to behave differently that the typical toArray() where an ADT(abstract data type) is converted to Array.
Maybe you can name it as toSaveFormat() ?

jazerler replied:
Good point. I've renamed it to `toSaveFormatArray` for now.

jazerler committed:
Rename methods and remove unused files

The sceneplayground package and the related resources were
no longer being used.
The methods toArray in Task have been renamed to toSaveFormatArray.

jazerler committed:
Fix checkstyle iissues

jazerler committed:
Add assertions to check Command inputs

jazerler committed:
Merge branch 'branch-A-Assertions'

jazerler committed:
Revert "Fix checkstyle iissues"

This reverts commit 12550976e2b1cd8802425945f27ecfab5e9be9d0.

jazerler committed:
Refactor code and improve GUI

jazerler committed:
Add assertion to Storage

jazerler committed:
Merge branch 'master' into branch-A-Assertions

jazerler committed:
Merge pull request #1 from jazerler/branch-A-Assertions

Add assertion to Storage

jazerler committed:
Merge pull request #2 from jazerler/branch-A-CodeQuality

Refactor code and improve GUI

jazerler committed:
Add workflow for CI

jazerler committed:
Merge pull request #3 from jazerler/branch-A-CI

Add workflow for CI

jazerler committed:
Create gradle.yml

jazerler committed:
Merge pull request #4 from jazerler/branch-A-CI

Create gradle.yml for CI

jazerler committed:
Fix issues with CI

jazerler committed:
Add help menu to commands

jazerler committed:
Merge pull request #5 from jazerler/branch-C-Help

Add help menu to commands

jazerler committed:
Change GUI colors and Duke picture

jazerler committed:
Change find results to show the actual indexes

Previously showed no indexes for both list and find.
Removed ResultsList since it was no longer being used.
Find results now show the index of Tasks in Storage's TaskList.

jazerler committed:
Change responses for commands and add readme

jazerler committed:
Update README.md

jazerler committed:
Update README
